src/pages/home/report/ReportActionsList.js

```diff
@@ -404,7 +405,31 @@ function ReportActionsList({
         if (!markerFound) {
             setCurrentUnreadMarker(null);
         }
-    }, [sortedVisibleReportActions, report.lastReadTime, report.reportID, messageManuallyMarkedUnread, shouldDisplayNewMarker, currentUnreadMarker]);
+    }, [sortedVisibleReportActions, shouldDisplayNewMarker, currentUnreadMarker, report.reportID]);
+
+    useEffect(() => {
+        calculateUnreadMarker();
+    }, [calculateUnreadMarker, report.lastReadTime, messageManuallyMarkedUnread]);
+
+    const visibilityCallback = useCallbackRef(() => {
```

Comment:
Why are we creating a callback ref here? @FitseTLT

Reply:
To solve closure problem, i.e if we pass the function directly as visibility callback the callback will be called with outdated prop from the component for instance `sortedVisibleReportActions`, `report` ...

Reply:
Was discussing the same on slack with contributor. I think we can just use `useCallback` with relevant dependency list to solve this issue instead of introducing a new hook.

Reply:
I recommend this [article](https://blog.stackademic.com/managing-function-re-creation-and-state-references-in-react-fefd8944939) for more context and the advantage of the use of the `useCallbackRef`

Reply:
The advantage is it makes the function's reference stable reduces the overhead of setting and clearing listeners, thus preventing performance degradation.

Reply:
My concern is it could have its own penality which we might not know of at the moment. Even the article could have missed it.
Similar to how we are saying `useCallback` vs `inline functions` in our contributing [guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#should-i-wrap-all-my-inline-functions-with-usecallback-or-move-them-out-of-the-component-if-they-have-no-dependencies) as a debatable topic. I think the same could be applied here even if you say it has more optimization like the `useCallback` vs `useCallbackRef`
And I think for this particular case, the use of new hook is not worth the trade-off since even without the hook we might not have that much of performance impact. I think the new hook is over-optimization in this case IMO. So the use of well-established `useCallback` should be enough optimization in this case if we need optimisation AFAIK. 


What do you think @hayata-suenaga ?

Reply:
I agree with @abdulrahuman5196 let's stick with `useCallback` for this case, but I really appreciate you taking into account performance @FitseTLT